src/multi-select/header.tsx

```diff
@@ -7,16 +7,23 @@ export const DropdownHeader = () => {
 
   const noneSelected = value.length === 0;
   const allSelected = value.length === options.length;
-  const customText = valueRenderer && valueRenderer(value, options);
+  const selectedText = valueRenderer
+    ? valueRenderer(value, options)
+    : "Text Undefined";
 
-  const getSelectedText = () => value.map((s) => s.label).join(", ");
-
-  return noneSelected ? (
-    <span className="gray">{customText || t("selectSomeItems")}</span>
-  ) : (
-    <span>
-      {customText ||
-        (allSelected ? t("allItemsAreSelected") : getSelectedText())}
-    </span>
+  const getSelectedText = useMemo(
+    () => () => value.map((s) => s.label).join(", "),
+    [value]
   );
+
+  switch (true) {
+    case noneSelected:
+      return (
+        <span className="gray">{selectedText || t("selectSomeItems")}</span>
+      );
+    case allSelected:
+      return <span>{selectedText || t("allItemsAreSelected")}</span>;
+    default:
+      return <span>{selectedText || getSelectedText()}</span>;
+  }
```

Comment:
Instead of using a ternary operator to determine the text to display, consider using a switch statement. This would make the code easier to read and understand, and would allow you to add additional cases more easily.